From e0c373c6dd9ff77e5eab862299fff4042e027194 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Tue, 16 Oct 2012 11:01:29 +0200
Subject: [PATCH] systemctl: append .service when unit does not have valid
 suffix

systemctl status a and systemctl status a.service lead to same output but
systemctl status a.b and systemctl status a.b.service do not.
(cherry picked from commit 696c245a23d55e4249651573eb9c61b68e61580c)
---
 src/shared/unit-name.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/shared/unit-name.c b/src/shared/unit-name.c
index 90f5842..0f3c571 100644
--- a/src/shared/unit-name.c
+++ b/src/shared/unit-name.c
@@ -469,7 +469,6 @@ char *unit_dbus_path_from_name(const char *name) {
 char *unit_name_mangle(const char *name) {
         char *r, *t;
         const char *f;
-        bool dot = false;
 
         assert(name);
 
@@ -491,10 +490,6 @@ char *unit_name_mangle(const char *name) {
                 return NULL;
 
         for (f = name, t = r; *f; f++) {
-
-                if (*f == '.')
-                        dot = true;
-
                 if (*f == '/')
                         *(t++) = '-';
                 else if (!strchr("@" VALID_CHARS, *f))
@@ -503,7 +498,7 @@ char *unit_name_mangle(const char *name) {
                         *(t++) = *f;
         }
 
-        if (!dot)
+        if (unit_name_to_type(name) < 0)
                 strcpy(t, ".service");
         else
                 *t = 0;
